tree/ntuple/src/RNTupleDescriptor.cxx

```diff
@@ -337,6 +337,24 @@ ROOT::NTupleSize_t ROOT::RNTupleDescriptor::GetNElements(ROOT::DescriptorId_t ph
    return result;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Return the cluster boundaries for each cluster in this RNTuple.
+std::vector<ROOT::RNTupleDescriptor::RClusterBoundaries> ROOT::RNTupleDescriptor::GetClusterBoundaries() const
+{
+   ROOT::RNTupleDescriptor::RClusterBoundaries clusterBoundaries;
+   std::vector<ROOT::RNTupleDescriptor::RClusterBoundaries> clusterBoundariesVec;
+   clusterBoundariesVec.reserve(GetNClusters());
+   auto clusterId = FindClusterId(0, 0);
+   while (clusterId != ROOT::kInvalidDescriptorId) {
+      const auto &clusterDesc = GetClusterDescriptor(clusterId);
+      clusterBoundaries.fFirstEntry = clusterDesc.GetFirstEntryIndex();
+      clusterBoundaries.fLastEntry = clusterDesc.GetFirstEntryIndex() + clusterDesc.GetNEntries();
+      clusterBoundariesVec.emplace_back(clusterBoundaries);
```

Comment:
Probably this could also work
```suggestion
   std::vector<ROOT::RNTupleDescriptor::RClusterBoundaries> clusterBoundariesVec;
   clusterBoundariesVec.reserve(GetNClusters());
   auto clusterId = FindClusterId(0, 0);
   while (clusterId != ROOT::kInvalidDescriptorId) {
      const auto &clusterDesc = GetClusterDescriptor(clusterId);
      clusterBoundariesVec.emplace_back({clusterDesc.GetFirstEntryIndex(), clusterDesc.GetFirstEntryIndex() + clusterDesc.GetNEntries()});
```